_RFCs/0003-tracking-package-versions.md

```diff
@@ -0,0 +1,56 @@
+
+However, we do not have a process in place to handle edge releases of software. As of today, Docker 18 is the latest edge version of Docker, but the latest stable version is still Docker 17. This brought us to the question - should "core/docker" track the latest stable version or the latest edge version?
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Many Habitat packages track and package 3rd party software - this includes Docker, Node, Java, and more. We learned early on that even when a new major version of the software was released, many people still used the older version (and sometimes did not have the choice of upgrading immediately due to compliance or other reasons).
+
+Historically, when a new major version of software was released (i.e. when Node 8 became the new LTS over Node 7) we changed the generic package (i.e. core/node) to track the newest major version and, for people who needed to use an older major version of the software, created a new plan (i.e. "core/node7") that they could refer to.
+
+Now we will create a new core plan for every major release of a piece of software, edge or stable (i.e. when Docker11 is released as edge, we will create a core/docker11 plan so that someone can use the edge version if they wish).
+
+Generic packages (i.e core/docker) will continue to track the latest stable version of the software. However, they will do this by sourcing whatever version plan corresponds to the latest stable version (i.e. if Docker10 is stable, core/docker will source core/docker10, even if core/docker11 tracks the edge version).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+This RFC would not change any current plans, it would apply to updated and new plans from the time of the adoption of the RFC.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This does potentially introduce many more plans into core/plans - which will need to be maintained and rebuilt as necessary, and it will increase the build load on Builder.
+
+# Rationale and alternatives
+[alternatives]: #alternatives
+
+The alternative is to keep doing what we have historically done - having the generic plan (i.e. core/node) track the latest major stable version of the software, and add in core/nodeX whenever a new major version of Node is released to stable).
+
+However, this would prevent our users from using the edge versions of software should they wish to.  It also makes it more difficult for users to "pin" to a certain version of software if they know they are not going to be upgrading to the next major version in the near future.
```

Comment:
Hmm I'm not 100% sure I follow this - could you clarify further? Specifically I'm not sure what part makes it harder for folks to pin.

Reply:
This looks like it was in response to one of the concerns in my message above.  Namely, that without major-version packages for the most recent stable release, there is no way to track minor releases on that major series without also having to eat a major-version upgrade at some point.

Reply:
Ah I see. So where does the line on this get drawn? E.G. Do we feel that adding major version packages for any inbound software is a reasonable support burden? If not, I believe the current process calls for creating major versions for packages as currently supported versions roll off but only in the case that the older version is still supported. Do we instead go the path of just _always_ creating a major version package when a version rolls off and do this by attrition?

Reply:
I believe adding major version packages for any new inbound software is a reasonable support burden if it cuts down on confusion.  I'm inclined to go with always creating a major version package when a version rolls off.  What we should do with versions that are no longer supported is important, but I feel that should be a separate RFC around when do we remove core plans and how.